src/s3/commands/createBucket.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getLogger } from '../../shared/logger'
+import { S3Node } from '../explorer/s3Nodes'
+import { DefaultCommands, Commands } from '../../shared/vscode/commands'
+import { DefaultWindow, Window } from '../../shared/vscode/window'
+import * as telemetry from '../../shared/telemetry/telemetry'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+
+/**
+ * Creates a bucket in the s3 region represented by the given node.
+ *
+ * Prompts the user for the bucket name, applying basic validations.
+ * Creates the bucket.
+ * Refreshes the node.
+ */
+export async function createBucketCommand(
+    node: S3Node,
+    window: Window = new DefaultWindow(),
+    commands: Commands = new DefaultCommands()
+): Promise<void> {
+    getLogger().debug(`CreateBucket called for ${node}`)
+
+    const bucketName = await window.inputBox.show({
```

Comment:
Thanks, looked into it. Nice that we have an abstraction to treat the two basically the same. Definitely makes sense for stuff with steps or more complex behavior. In this case, looks like the only added value is the title. Sticking with the simpler form since the intent seems self explanatory.

Reply:
We've found out that earlier experiences were more confusing when they didn't have a title.

Reply:
Lets revisit this and see if it's worth sticking to one or the other. I'm seeing mixed usage throughout the code right now.